slice/internal/core/core.go

```diff
@@ -20,12 +20,12 @@ import (
 	"regexp"
 
 	corev1 "k8s.io/api/core/v1"
-
-	"tpu-slice-controller/api/v1alpha1"
 )
 
-var SliceStates = []v1alpha1.SliceConditionType{
-	v1alpha1.Error, v1alpha1.Deformed, v1alpha1.Forming, v1alpha1.Degraded, v1alpha1.Ready,
+var SliceStates = []MMIGHealthStatus{
+	MMIGHealthStatusActivating, MMIGHealthStatusActive, MMIGHealthStatusActiveDegraded,
+	MMIGHealthStatusDeactivating, MMIGHealthStatusFailed, MMIGHealthStatusIncomplete,
+	MMIGHealthStatusUnknown,
```

Comment:
I think we should keep separate the notion of slice state from MMIGHealthStatus. This fewer number of states will be more managable in the controller. WDYT? 

We would have a function of Ready condtion's (Status, Reason) -> SliceState.

e.g.

```
(Unknown, "") -> ACCEPTED
(False, Incomplete) -> INITIALIZING
(False, Activating) -> INITIALIZING
...
(False, Failed) -> ERROR
```